app/src/pages/management/Login.svelte

```diff
@@ -232,12 +239,65 @@
 			}
 		}
 	};
+
+	let notificationModalOpen = false;
+
+	$: {
+		if ((user.token)) {
+			console.log("I have a token");
+			console.log(user);
+			console.log(Notification.permission);
+			if ((!(Notification.permission === "granted")) && !settings.notificationsDoNotAsk) {
+				console.log("here 1")
+				notificationModalOpen = true;
+			} else if ((!(Notification.permission === "granted") && settings.notificationsDoNotAsk) || Notification.permission === "granted") {
+				console.log("here 2")
+				notificationModalOpen = false;
+			}
+		} else {
+			console.log("here 3")
+			notificationModalOpen = false;
+		}
+	}
 </script>
 
 <svelte:head>
 	<script src="https://accounts.google.com/gsi/client" async></script>
 </svelte:head>
 
+<Modal bind:open={notificationModalOpen} size="sm" dialogClass="fixed top-0 start-0 end-0 h-modal md:inset-0 md:h-full z-40 w-full p-4 flex">
+	<h1 class="font-bold text-xl">Enable Notifications</h1>
+	<h2>Enable to get notifications for Tickets, and/or when a robot loses connection during a match</h2>
+	<Button
+		color="primary"
+		class="w-fit"
+		size="sm"
+		on:click={() => {
+			try {
+				Notification.requestPermission().then((result) => {
+					if (result === "granted") {
+						$settingsStore.notifications = true;
+						subscribeToPush();
+					}
+				});
+				notificationModalOpen = false;
+			} catch (e) {
+				console.error(e);
+				toast("Error", "Error requesting notification permissions");
+			}
+		}}>Grant Notification Permissions</Button
+	>
+	<Button color="primary" class="w-fit" size="sm" on:click={() => {
+		notificationModalOpen = false;
+	}}>No, Thank You</Button>
+	<Button color="primary" class="w-fit" size="sm" on:click={() => {
+		settings.notificationsDoNotAsk = true;
+		notificationModalOpen = false;
+	}}>Do Not Ask Again</Button>
+	<p class="text-sm">You can change which types of notifications you are subscribed to from the Settings screen</p>
+	<p class="text-sm">***You cannot close this window without selecting an option***</p>
```

Comment:
Let's make it so when you click X or outside the modal it treats it like saying "No, Thank You"